config/options

```diff
@@ -4,77 +4,73 @@ if [[ $EUID -eq 0 ]]; then
 ROOT="${PWD}"
-DISTRO_DIR="$ROOT/distributions"
-PROJECT_DIR="$ROOT/projects"
+DISTRO_DIR="${ROOT}/distributions"
+PROJECT_DIR="${ROOT}/projects"
 
 # determines DISTRO, if not forced by user
 # default is LibreELEC
-if [ -z "$DISTRO" ]; then
-  DISTRO="LibreELEC"
-else
-  DISTRO="$DISTRO"
-fi
+DISTRO="${DISTRO:-LibreELEC}"
 
 # determines PROJECT, if not forced by user
 # default is Generic
-if [ -z "$PROJECT" ]; then
-  PROJECT="Generic"
-else
-  PROJECT="$PROJECT"
-fi
+PROJECT="${PROJECT:-Generic}"
 
-# determines TARGET_ARCH, if not forced by user (x86_64 / arm)
+# determines TARGET_ARCH, if not forced by user
 # default is x86_64
-if [ -z "$ARCH" ]; then
-  TARGET_ARCH="x86_64"
-else
-  TARGET_ARCH="$ARCH"
-fi
+ARCH="${ARCH:-x86_64}"
+TARGET_ARCH="${ARCH}"
 
 # include helper functions
-. config/functions
+  . config/functions
 
 # include versioning
   . config/version
 
-# read distro versioning if available
- if [ -f "$DISTRO_DIR/$DISTRO/version" ]; then
-   . $DISTRO_DIR/$DISTRO/version
- fi
+# read DISTRO versioning if available
+  if [ -f "${DISTRO_DIR}/${DISTRO}/version" ]; then
+    . ${DISTRO_DIR}/${DISTRO}/version
+  fi
 
-# read distro options if available
- if [ -f "$DISTRO_DIR/$DISTRO/options" ]; then
-   . $DISTRO_DIR/$DISTRO/options
- fi
+# read DISTRO options if available
+  if [ -f "${DISTRO_DIR}/${DISTRO}/options" ]; then
+    . ${DISTRO_DIR}/${DISTRO}/options
+  fi
 
-# read project options if available
- if [ -f "$PROJECT_DIR/$PROJECT/options" ]; then
-   . $PROJECT_DIR/$PROJECT/options
- fi
+# read PROJECT options if available
+  if [ -f "${PROJECT_DIR}/${PROJECT}/options" ]; then
+    . ${PROJECT_DIR}/${PROJECT}/options
+  fi
 
-# read board options if available
- if [ -f "$PROJECT_DIR/$PROJECT/devices/$DEVICE/options" ]; then
-   . $PROJECT_DIR/$PROJECT/devices/$DEVICE/options
+# read DEVICE options if available
+ if [ -f "${PROJECT_DIR}/${PROJECT}/devices/${DEVICE}/options" ]; then
+   . ${PROJECT_DIR}/${PROJECT}/devices/${DEVICE}/options
  fi
 
 # projects can set KERNEL_NAME (kernel.img)
-[ -z "$KERNEL_NAME" ] && KERNEL_NAME="KERNEL"
+KERNEL_NAME="${KERNEL_NAME:-KERNEL}"
+
+LINUX_DEPENDS="${PROJECT_DIR}/${PROJECT}/linux ${PROJECT_DIR}/${PROJECT}/patches/linux \
+  ${PROJECT_DIR}/${PROJECT}/packages/linux \
+  ${ROOT}/packages/linux"
 
-LINUX_DEPENDS="$PROJECT_DIR/$PROJECT/linux $PROJECT_DIR/$PROJECT/patches/linux $PROJECT_DIR/$PROJECT/packages/linux $ROOT/packages/linux"
-[ -n "$DEVICE" ] && LINUX_DEPENDS+=" $PROJECT_DIR/$PROJECT/devices/$DEVICE/linux $PROJECT_DIR/$PROJECT/devices/$DEVICE/patches/linux $PROJECT_DIR/$PROJECT/devices/$DEVICE/packages/linux"
-[ "$TARGET_ARCH" = "x86_64" ] && LINUX_DEPENDS+=" $ROOT/packages/linux-firmware/intel-ucode $ROOT/packages/linux-firmware/kernel-firmware"
+[ -n "${DEVICE}" ] && LINUX_DEPENDS+=" ${PROJECT_DIR}/${PROJECT}/devices/${DEVICE}/linux \
+  ${PROJECT_DIR}/${PROJECT}/devices/${DEVICE}/patches/linux \
+  ${PROJECT_DIR}/$PROJECT/devices/${DEVICE}/packages/linux"
```

Comment:
We should agree what the "standard" is (with or without braces) as a team before mass changing, otherwise we become increasingly inconsistent with other files using the old standard. Once we agree what the standard is then we can mass change, or change as we go.

I would say the current "standard" (based on existing code) is to **not** use braces unless they are required (string substitution etc.).

We should discuss this more, either here or Slack.

Reply:
In PR3055 I've added that we should use `${FOO}` after a vote (currently 3 for, 1 against, 1 smiley).